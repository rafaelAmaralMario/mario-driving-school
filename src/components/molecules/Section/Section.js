import React from "react";
import PropTypes from "prop-types";
import styled, { css } from 'styled-components';

import { Container } from 'components/atoms';

export const Content = styled.div`
  padding: 40px 0;
  ${props => props.inverted && css`
      background-color: #f7f7f7;
    `
  }
  h2 {
    margin-top: 0;
  }
`;

const Section = ({ inverted, children }) => (
  <Content inverted={inverted}>
    <Container>
      {children}
    </Container>
  </Content>
);

Section.defaultProps = {
  children: undefined,
  inverted: false
}

Section.propTypes = {
  children: PropTypes.node,
  inverted: PropTypes.bool

}

export default Section;
