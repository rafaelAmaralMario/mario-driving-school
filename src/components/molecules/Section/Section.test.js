import React from 'react'
import { render } from 'utils/test-utils';

import Section from './Section';

test('should match with snapshot', () => {
  const { asFragment } = render(<Section></Section>);
  expect(asFragment()).toMatchSnapshot();
})
