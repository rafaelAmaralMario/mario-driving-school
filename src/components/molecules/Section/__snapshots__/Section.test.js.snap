// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<DocumentFragment>
  .c1 {
  width: 100%;
  padding: 0 8px;
}

.c0 {
  padding: 40px 0;
}

.c0 h2 {
  margin-top: 0;
}

@media (min-width:undefinedpx) {
  .c1 {
    padding: 0 16px;
  }
}

@media (min-width:992px) {
  .c1 {
    max-width: 1140px;
    margin: 0 auto;
    padding: 0;
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
    />
  </div>
</DocumentFragment>
`;
